src/rp2_common/pico_standard_link/CMakeLists.txt

```diff
@@ -78,6 +78,99 @@ if (NOT TARGET pico_standard_link)
         set_target_properties(${TARGET} PROPERTIES PICO_TARGET_BINARY_TYPE ${TYPE})
     endfunction()
 
+    # pico_set_modified_binary_type(TARGET TYPE [RAM <RAM_ORIGIN> <RAM_LENGTH>] [SCRATCH_X <SCRATCH_X_ORIGIN> <SCRATCH_X_LENGTH>] [SCRATCH_Y <SCRATCH_Y_ORIGIN> <SCRATCH_Y_LENGTH>])
+    # \brief\ Set the binary type for the target with modified RAM, SCRATCH_X and SCRATCH_Y regions
+    #
+    # \param\ TYPE The type of binary to use (flash or sram)
+    # \param\ RAM_ORIGIN The origin of the RAM section
+    # \param\ RAM_LENGTH The length of the RAM section
+    # \param\ SCRATCH_X_ORIGIN The origin of the SCRATCH_X section
+    # \param\ SCRATCH_X_LENGTH The length of the SCRATCH_X section
+    # \param\ SCRATCH_Y_ORIGIN The origin of the SCRATCH_Y section
+    # \param\ SCRATCH_Y_LENGTH The length of the SCRATCH_Y section
+    function(pico_set_modified_binary_type TARGET TYPE)
+        set(multiValueArgs RAM SCRATCH_X SCRATCH_Y)
+        cmake_parse_arguments(PARSE_ARGV 0 args
+            "" "" "${multiValueArgs}"
+        )
+
+        if (NOT (TYPE STREQUAL "sram" OR TYPE STREQUAL "flash"))
+            message(FATAL_ERROR "Invalid binary type for pico_set_modified_binary_type: ${TYPE}")
+        endif()
+
+        # Default RAM, SCRATCH_X and SCRATCH_Y regions
+        if (PICO_RP2040)
+            set(PICO_DEFAULT_RAM_ORIGIN 0x20000000)
+            set(PICO_DEFAULT_RAM_LENGTH 256k)
+            set(PICO_DEFAULT_SCRATCH_X_ORIGIN 0x20040000)
+            set(PICO_DEFAULT_SCRATCH_X_LENGTH 4k)
+            set(PICO_DEFAULT_SCRATCH_Y_ORIGIN 0x20041000)
+            set(PICO_DEFAULT_SCRATCH_Y_LENGTH 4k)
+        elseif (PICO_RP2350)
+            set(PICO_DEFAULT_RAM_ORIGIN 0x20000000)
+            set(PICO_DEFAULT_RAM_LENGTH 512k)
+            set(PICO_DEFAULT_SCRATCH_X_ORIGIN 0x20080000)
+            set(PICO_DEFAULT_SCRATCH_X_LENGTH 4k)
+            set(PICO_DEFAULT_SCRATCH_Y_ORIGIN 0x20081000)
+            set(PICO_DEFAULT_SCRATCH_Y_LENGTH 4k)
+        endif()
+
+        # Scripts that will be created by this function
+        set(LINKER_CMAKE_SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/memmap_${TARGET}.cmake")
+        set(LINKER_LD_SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/memmap_${TARGET}.ld")
+
+        # Configure memmap_${TARGET}.cmake script file to create memmap_${TARGET}.ld
```

Comment:
Is it worth having this function checking that none of the LENGTH values are too big for the given chip? (i.e. I guess you could set a 300KB RAM_LENGTH for RP2350, but not for RP2040?)

Reply:
That check would only be useful on RP2040 to ensure the length is <264k, as on RP2350 you could be using PSRAM so can have a length of 16M - but that should be caught anyway because if you set a RAM region >256k on RP2040 but have the correct SCRATCH regions, then they'll overlap and throw a linker error. So I think I'll probably not add this check, and leave it to the linker